crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -1107,6 +1099,7 @@ impl<T: PoolTransaction> AllTransactions<T> {
                 if cumulative_cost > on_chain_balance {
                     // sender lacks sufficient funds to pay for this transaction
                     tx.state.remove(TxState::ENOUGH_BALANCE);
+                    return Err(InsertErr::Overdraft { transaction })
```

Comment:
this now returns an error but the tx remains in the set
this is also not necessarily the same tx that we inserted.

How does geth handle descendant txs that are also in overdraft?